docs: IA calibration (P0): add Tune "Noise & relevance" and "Release timing" pages (#705)
Promptless/docs: src/content/docs/docs/tune/release-timing.mdx

Reviewer: We should be referring to UI strings/flows generally, not YAML configs. Most users are going to be using the UI. Remember this as a preference for this doc collection.

---
title: Time suggestions to your release cycle
description: >-
  Match when Promptless drafts and publishes documentation to how your product
  ships, from pull request to feature-flag rollout
slug: docs/tune/release-timing
sidebar:
  hidden: false
  order: 23
  label: Release timing
---
import Note from '@components/fern/Note.astro';
import Tip from '@components/fern/Tip.astro';
import Info from '@components/fern/Info.astro';

> A documentation suggestion is only useful when it arrives at the right moment. Draft it too early and you document work your reviewers haven't finished; draft it too late and the docs trail the release. To get the timing right, align the event that triggers Promptless — and the moment it publishes — with the point in your release cycle when the change becomes real.

Reviewer: ```suggestion
A documentation suggestion is only useful when it arrives at the right moment. Draft it too early and you document work your reviewers haven't finished. Draft it too late and the docs trail the release. To get the timing right, align the event that triggers Promptless — and the moment docs publish — with the point in your release cycle when the change becomes real.
```

There's no separate "timing" setting to configure. You control timing by choosing which event triggers Promptless and how eagerly it publishes, using the trigger and policy fields you already have.

## Match the trigger event to your release stage

The `trigger_on` event on a GitHub PR trigger decides how finished a change is when Promptless analyzes it. Each event maps to a stage in the change's life:

| Trigger event | Fires when | Choose it when |
|---|---|---|
| `opened` | A pull request opens | You want documentation drafted alongside the code, so reviewers evaluate both together. |
| `first_approval` | A pull request gets its first approval | You want to wait until a reviewer has vetted the change, so Promptless analyzes reviewed work rather than a work in progress. |

> | `merge` | A pull request merges | You only want documentation for changes that have actually shipped. |

Reviewer: ```suggestion
| `merge` | A pull request merges | You only want documentation for changes that have actually made it into the code base. |
```

```yaml
triggers:
  reviewed-changes:
    trigger_type: github_pr
    match:
      - repos:
          - acme/backend
        trigger_on:
          - first_approval
```

> If suggestions keep arriving before a change has settled, move the trigger later — from `opened` to `first_approval`, or to `merge`. List more than one event when you want both early feedback and a post-merge refresh. See [GitHub PRs → Trigger modes](/docs/connect/triggers/github-prs#trigger-modes) for the full behavior of each mode.

Reviewer: ```suggestion
If suggestions keep arriving before a change has settled, move the trigger from `opened` to `first_approval`, or from `first_approval` to `merge`. List more than one event when you want both early feedback and a post-merge refresh. See [GitHub PRs → Trigger modes](/docs/connect/triggers/github-prs#trigger-modes) for the full behavior of each mode.
```

<Info>
`first_approval` fires once per pull request, on the first approval only. Later approvals or re-approvals after requested changes don't trigger another run. If you merge your own pull requests without a separate reviewer, `first_approval` never fires — use `merge` or a commit trigger instead.
</Info>

## Document only what has shipped

To track shipped changes rather than proposed ones, add a [commit trigger](/docs/connect/triggers/github-commits), which fires when commits land on a monitored branch. Pair it with your PR trigger to get both timings: Promptless comments on pull requests while they're open, then refreshes the suggestion with the final content after the change merges to your default branch.

```yaml
triggers:
  main-commits:
    trigger_type: github_commit
    match:
      - repos:
          - acme/backend
        branches:
          - main
```

Using commits on their own — without a PR trigger — is the quietest option. Promptless runs only after changes reach your default branch, so it fires less often.

## Control when a suggestion publishes

> Trigger timing decides when Promptless *drafts*; publishing policies decide when it *ships*. By default, a suggestion waits for a human to review and publish it. When a class of change should publish without that wait, set `auto_merge` in a policy rule so the documentation PR merges as soon as it's created.

Reviewer: ```suggestion
Trigger timing decides when Promptless *drafts*, but publishing policies decide when it *ships*. By default, a suggestion waits for a human to review and publish it. When a class of change should publish without that wait, set `auto_merge` in a policy rule so the documentation PR merges as soon as it's created.
```

```yaml
policies:
  rules:
    - if:
        trigger: main-commits
      then:
        publishing:
          auto_create_pr: true
          auto_merge: true
```

Auto-merge requires automatic PR creation, and it publishes without a human review step — the documentation PR merges on creation, so no one intercepts it first. That fits high-confidence workflows — internal documentation, or changelog updates you want live the moment the change lands. Scope it to a specific trigger so it applies only where you trust full automation.

<Note>
Publishing policies resolve at publish time, not when the trigger fires. If you change a policy while a suggestion is being drafted, the new values apply when that suggestion is published.
</Note>

## Trigger from a feature-flag rollout

> When a feature ships behind a flag, the pull request can merge weeks before users see the change — so PR- and commit-based timing may draft the docs long before they're relevant. The [LaunchDarkly integration](/docs/reference/integrations/launchdarkly) closes that gap. Once connected, Promptless watches for feature-flag changes in your LaunchDarkly projects and starts documentation work when a flag changes, rather than when the code merged.

Reviewer: ```suggestion
When a feature ships behind a flag, the pull request can merge weeks before users see the change, so PR- and commit-based timing may draft the docs long before they're relevant. The [LaunchDarkly integration](/docs/reference/integrations/launchdarkly) closes that gap. Once connected, Promptless watches for feature-flag changes in your LaunchDarkly projects and starts documentation work when a flag changes, rather than when the code merged.
```

> That makes the flag rollout — not the merge — the moment documentation updates, matching how flagged features actually reach users.

Reviewer: ```suggestion
That makes the flag rollout, not the merge, the moment documentation updates, matching how flagged features actually reach users.
```

<Note>

> The LaunchDarkly integration is in beta. Connect it from the [integrations page](https://app.gopromptless.ai/integrations); see the [LaunchDarkly integration guide](/docs/reference/integrations/launchdarkly) for the setup and permissions it needs.

Reviewer: ```suggestion
The LaunchDarkly integration is in beta. Connect it from the [integrations page](https://app.gopromptless.ai/integrations). See the [LaunchDarkly integration guide](/docs/reference/integrations/launchdarkly) for the setup and permissions it needs.
```
